discretize/tests.py

```diff
@@ -633,6 +637,110 @@ def Quadratic(x, return_g=True, return_H=True):
     return Quadratic
 
 
+# TESTS
+def dottest(
```

Comment:
how about we rename this function to `assert_isadjoint` ? Maybe follow the `np.testing` naming scheme for it.

Reply:
Done, good idea.